esphomeRemote/esphomeRemoteLight.h

```diff
@@ -2,30 +2,137 @@
+    std::string color_mode = "";
     bool onState;
 
+    // TODO:
+    // * change slider design to match overall design
+    // * only show sliders if the light supports them 
+    // * reduce number of functions (maybe just one which is handed a struct
+    // * cleanup
+    // * only show lightDetailMenu if light supports it. Otherwise just toggle it
+    // * only enable scrolling menu if light is turned on
+    // * if light is off show sliders as inactive
+
+    void decTemperature() {
+        std::stringstream ss;
+        ss << (color_temp - 10);
+        const std::map< std::string, std::string > data = {
+                                                              {"entity_id",entityId.c_str()},
+                                                              {"color_temp", ss.str()} ,
+                                                          };
+        setAttribute(data);
+    }
+
+    void incTemperature() {
+        std::stringstream ss;
+        ss << (color_temp + 10);
+        const std::map< std::string, std::string > data = {
+                                                              {"entity_id",entityId.c_str()},
+                                                              {"color_temp", ss.str()} ,
+                                                          };
+        setAttribute(data);
+    }
+
+    void decBrightness() {
+        std::stringstream ss;
+        ss << (brightness - 10);
+        const std::map< std::string, std::string > data = {
+                                                              {"entity_id",entityId.c_str()},
+                                                              {"brightness", ss.str()} ,
+                                                          };
+        setAttribute(data);
+    }
+
+    void incBrightness() {
+        std::stringstream ss;
+        ss << (brightness + 10);
+        const std::map< std::string, std::string > data = {
+                                                              {"entity_id",entityId.c_str()},
+                                                              {"brightness", ss.str()} ,
+                                                          };
+        setAttribute(data);
+    }
+
+    void setAttribute(const std::map< std::string, std::string> &data){
+        call_homeassistant_service("light.turn_on", data);
+    }
+
     void toggleLight() {
       call_homeassistant_service("light.toggle", {
         {"entity_id", entityId.c_str()}
       });
     }
 
+    bool supportsBrightness(){
+        return color_mode.compare("onoff") == 0 || color_mode.compare("unknown") == 0;
+    }
+
+    bool supportsColorTemperature(){
+        return false;
+    }
+
   private:
     void state_changed(std::string newOnState) {
-      ESP_LOGI("Light", "state changed to %s", newOnState.c_str());
+      ESP_LOGI("brightness", " changed to %s", newOnState.c_str());
+      ESP_LOGI("color_mode", "state changed to %s", color_mode.c_str());

```

Comment:
this works pretty well but it might be better to provide this in the light setup on compile so we can subscribe to fewer states while running

Reply:
there we can also specify if the light supports rgb or temperature

Reply:
nevermind all of this. colorMode works very well for this and it's only 1 state. nice work